aodndata/aatams/aatams_sattag.py

```diff
@@ -97,42 +146,34 @@ def schedule_file_removal(self, remote_pipeline_file):
         )
 
     def process_nrt(self):
-        """Process NRT files, only allowing updates to occur if the new files for certain
-        campaigns are more recent."""
+        """Process NRT files, only allowing updates to occur if the new files for
+        certain campaigns are more recent."""
         previous_files = self.state_query.query_storage(self.dest_path_function(""))
         if not previous_files:
             return
 
-        all_campaigns = set(self.get_campaigns(self.file_collection))
-        more_than_one_campaign = len(all_campaigns) > 1
-        if more_than_one_campaign:
-            raise InvalidFileContentError(
-                NRT_ZIPFILE_WITH_DIFF_CAMPAIGNS.format(
-                    self._file_basename, all_campaigns
-                )
-            )
+        current_campaign = self.get_current_campaign()
+        if not current_campaign:
+            return
 
-        campaign = all_campaigns.pop()
-        old_metadata_file = self.get_metadata_file(previous_files, campaign)
+        try:
+            old_zip_file = [x for x in previous_files if current_campaign in x.name][0]
+        except IndexError:
+            return
+
+        old_metadata_file = self.get_remote_metadata_from_zip(old_zip_file)
         if not old_metadata_file:
             return
 
-        new_metadata_file = self.get_metadata_file(self.file_collection, campaign)
+        new_metadata_file = self.get_metadata_file(
+            self.file_collection, current_campaign
+        )
 
         logger.info(
             NRT_TIMESTAMP_COMPARISON_MSG.format(
-                old_metadata_file.dest_path, new_metadata_file.src_path
+                old_zip_file.dest_path, new_metadata_file.src_path
             )
         )
-        # TODO: remove this compatibility layer when aodncore is updated
-        try:
-            self.state_query.download(
-                RemotePipelineFileCollection(old_metadata_file), self.temp_dir
-            )
-        except AttributeError:
-            self.state_query._storage_broker.download(
-                RemotePipelineFileCollection(old_metadata_file), (self.temp_dir)
-            )
 
         if not self.is_nrt_update_required(old_metadata_file.local_path):
             raise InvalidFileContentError(
```

Comment:
Yeap, good catch. I didn't add a test for this particular error msg, although I test another specific error msg.

This is now happening because the metadata file is not stored on s3 anymore. Hence, we lost the ```dest_path``` property. 

Fix is quick - just need to point to the old_zip_file it's comparing against and add a specific test for this msg.